src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTestBase.cs

```diff
@@ -276,7 +276,8 @@ public static class EditAndContinueTestExtensions
     {
         internal static CSharpCompilation WithSource(this CSharpCompilation compilation, CSharpTestSource newSource)
         {
-            return compilation.RemoveAllSyntaxTrees().AddSyntaxTrees(newSource.GetSyntaxTrees(TestOptions.Regular));
+            var previousParseOptions = compilation.SyntaxTrees.FirstOrDefault()?.Options as CSharpParseOptions;
```

Comment:
>as CSharpParseOptions [](http://example.com/codeflow?start=89&length=21)

Is this cast expected to fail, i.e. it is expected to encounter a value of a different type? Otherwise, please use regular type cast syntax.

Reply:
Changed, please check 1eed89b.